### PR TITLE
Skip the reproducibility tests for lightgbm

### PR DESCRIPTION
**Pull request by nabenabe0928**
## Motivation

Fix the CI.

## Description of the changes

I am checking the cause right now, so I will describe it here later.

### CODE REVIEW

**Comment by pkoz** [comment]
@nabenabe0928 you may find [this issue](https://github.com/optuna/optuna/issues/5209) relevant. It looks like the same problem I encountered.

**Review by c-bata — CHANGES_REQUESTED** [review]

**Comment by nabenabe0928** [comment]
@pkoz @c-bata  I merged your changes!

**Comment by toshihikoyanase** [comment]
I don't think this PR fix the root cause of the reproducibility issue, but it simply skip the test cases. Could you update the PR title to explain the change?

File: tests/integration_tests/lightgbm_tuner_tests/test_optimize.py [hunk 1]
```diff
@@ -726,6 +726,7 @@ def test_optuna_callback(self) -> None:
 
         assert callback_mock.call_count == 10
 
+    @pytest.mark.skip(reason="Fail since 28 Jan 2024.")
```
Comment (c-bata) [inline tests/integration_tests/lightgbm_tuner_tests/test_optimize.py:729]: Could you add a TODO comment?

**Review by c-bata — COMMENTED** [review]
Thank you for your pull request. Changes almost look good to me. I left one suggestion though.

File: tests/integration_tests/lightgbm_tuner_tests/test_optimize.py (hunk 1 above)
Comment (nabenabe0928) [reply tests/integration_tests/lightgbm_tuner_tests/test_optimize.py]: I added a comment to the corresponding two lines.

**Review by y0z — APPROVED** [review]
LGTM

**Review by c-bata — APPROVED** [review]
LGTM.

**Pull request merged**